src/Attribute/MapDocument.php

```diff
@@ -21,7 +21,8 @@ public function __construct(
         public array|string|null $id = null,
         bool $disabled = false,
         string $resolver = DocumentValueResolver::class,
+        public ?string $message = null,
     ) {
-        parent::__construct($class, $objectManager, $expr, $mapping, $exclude, $stripNull, $id, null, $disabled, $resolver);
+        parent::__construct($class, $objectManager, $expr, $mapping, $exclude, $stripNull, $id, null, $disabled, $resolver, $message);
```

Comment:
I think we can avoid locking in `^7.1` by checking if `$message` property exists in base class. We could also trigger a warning if someone passed the message when using older version of Symfony.

Reply:
Done. 
Not sure should it be `@trigger_error` or `RuntimeException` as trigger_error newer used at the project.

Reply:
`RuntimeException` is correct. Setting the `$message` argument was not possible previously, so nothing is being broken.